### ISSUE TITLE
Transition callbacks

### ISSUE THREAD

**Issue by dcosson** [aasm/aasm#371]
Looking at the callback orderings, it seems that transition callbacks are pretty inconsistent with event callbacks.

Event goes: before -> guards -> success -> after -> after_all_events

But Transition goes: guards -> after_all_transitions -> after -> success

It's seems logical that an `after_all` should come after `after`, and it seems potentially dangerous that `after` and `after_all` are both called before success meaning if, say, there's a temporary network issue and the db write fails, it could still trigger the `after` callbacks (which might write to an external service or something, leading to an inconsistent state).

Is it at all feasible to re-order transition callbacks to be consistent with how they're defined for event? (i.e. success -> after -> after_all). 

My immediate issue is that I want to be able to insert some logic after a transition, but before `after_all_transitions`. A less comprehensive but backwards compatible solution would be just to add an `after_guards` transition callback or something like that, which gets called right before `after_all_transitions`

**Comment by Ashviniv** [comment]
@dcosson I agree with you. One solution I would suggest you, is to fork the repo. and change the order of two lines in the [`lib/aasm/core/transition.rb`](https://github.com/Ashviniv/aasm/blob/e047f95f4c5e0e9e17ca410bedc5a92a203d097b/lib/aasm/core/transition.rb#L36)  and use the code, which will execute `after transition` first and then 'after_all_transitions'
I would like to send a pull request If I get a confirmation.

**Comment by dcosson** [comment]
Cool, that would be sweet if you have time for a PR. For now I'm just abusing guards to get a callback that fires after_all_transitions.

**Comment by Ashviniv** [comment]
@dcosson Submitted a PR #379.
